contracts/hooks/KintoHook.sol

```diff
@@ -107,21 +116,31 @@ contract KintoHook is LimitHook {
         address receiver = params_.transferInfo.receiver;
         address msgSender = abi.decode(params_.transferInfo.data, (address));
 
-        if (kintoFactory.walletTs(receiver) == 0)
-            revert InvalidReceiver(receiver);
-        if (!kintoID.isKYC(IKintoWallet(receiver).owners(0)))
-            revert KYCRequired();
-        if (!senderAllowlist[msgSender]) {
-            if (!IKintoWallet(receiver).isFunderWhitelisted(msgSender))
-                revert SenderNotAllowed(msgSender);
+        if (!(receiveAllowlist[receiver] && senderAllowlist[msgSender])) {
```

Comment:
This `if (!senderAllowlist[msgSender])` is never met right now.

I think you want something like this

```
if (!receiveAllowlist[receiver]) {
    if (kintoFactory.walletTs(receiver) == 0) {
        revert InvalidReceiver(receiver);
    }

    if (!kintoID.isKYC(IKintoWallet(receiver).owners(0))) {
        revert KYCRequired();
    }
    revert InvalidReceiver(receiver);
}

if (!senderAllowlist[msgSender] && !IKintoWallet(receiver).isFunderWhitelisted(msgSender)) {
        revert SenderNotAllowed(msgSender);
    }
}

```